rust/main/hyperlane-base/src/settings/aws_credentials.rs

```diff
@@ -17,34 +17,73 @@ use rusoto_sts::WebIdentityProvider;
 /// The primary use case is running Hyperlane agents in AWS Kubernetes cluster (EKS) configured
 /// with [IAM Roles for Service Accounts (IRSA)](https://aws.amazon.com/blogs/containers/diving-into-iam-roles-for-service-accounts/).
 /// The IRSA approach follows security best practices and allows for key rotation.
+/// 3) `InstanceMetadataProvider`: retrieves credentials from EC2 instance metadata service (IMDSv2).
+/// This allows EC2 instances with attached IAM instance profiles to authenticate without static credentials.
 pub(crate) struct AwsChainCredentialsProvider {
     environment_provider: EnvironmentProvider,
     web_identity_provider: AutoRefreshingProvider<WebIdentityProvider>,
+    instance_metadata_provider: AutoRefreshingProvider<InstanceMetadataProvider>,
 }
 
 impl AwsChainCredentialsProvider {
     pub fn new() -> Self {
         // Wrap the `WebIdentityProvider` to a caching `AutoRefreshingProvider`.
         // By default, the `WebIdentityProvider` requests AWS Credentials on each call to `credentials()`
         // To save the CPU/network and AWS bills, the `AutoRefreshingProvider` allows to cache the credentials until the expire.
-        let auto_refreshing_provider =
+        let web_identity_auto_refreshing =
```

Comment:
please dont change the existing code unnecessarily it will be harder to merge upstream back into our fork